common-security/src/main/java/co/cask/common/security/authorization/ACLStore.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright © 2014 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.common.security.authorization;
+
+import java.util.Set;
+
+/**
+ * Provides methods for storing and querying {@link ACLEntry}s.
+ */
+public interface ACLStore {
+
+  /**
+   * Writes a single {@link ACLEntry}.
+   *
+   * @param entry the {@link ACLEntry} to write
+   * @return true if entry did not previously exist
+   */
+  boolean write(ACLEntry entry);
```

Comment:
Is the boolean return value here used?  While this probably would not be a high volume operation, returning whether or not the value existed turns all write operations into read/write operations, with a higher cost.